app/src/main/java/tech/ula/MainActivity.kt

```diff
@@ -161,6 +170,65 @@ class MainActivity : AppCompatActivity(), SessionListFragment.SessionSelection,
         ACRA.init(applicationContext as Application, builder)
     }
 
+    private fun setupReviewRequest() {
+        val minimumTimePassedReached = usageUtility.getIsSufficientTimeElapsedSinceFirstOpen()
+        val minimumTimesOpenedReached = usageUtility.numberOfTimesOpenedIsGreaterThanThreshold()
+        val userAlreadyGaveReviewOrDismissed = usageUtility.getUserGaveFeedback()
+
+        if (minimumTimePassedReached && minimumTimesOpenedReached && !userAlreadyGaveReviewOrDismissed) {
```

Comment:
If you follow the suggestion below, you could change this to simply 
`if (feedbackUtility.askingForFeedbackIsAppropriate)  setupReviewRequestUI()`
and then you can move the rest of the UI setup there as well.